packages/flutter/lib/src/widgets/drag_target.dart

```diff
@@ -742,14 +742,14 @@ class _DragTargetState<T extends Object> extends State<DragTarget<T>> {
       _candidateAvatars.remove(avatar);
     });
     widget.onAccept?.call(avatar.data! as T);
-    widget.onAcceptWithDetails?.call(DragTargetDetails<T>(data: avatar.data! as T, offset: avatar._lastOffset!));
+    widget.onAcceptWithDetails?.call(DragTargetDetails<T>(data: avatar.data as T?, offset: avatar._lastOffset!));
   }
 
   void didMove(_DragAvatar<Object> avatar) {
     if (!mounted) {
       return;
     }
-    widget.onMove?.call(DragTargetDetails<T>(data: avatar.data! as T, offset: avatar._lastOffset!));
+    widget.onMove?.call(DragTargetDetails<T>(data: avatar.data as T?, offset: avatar._lastOffset!));
```

Comment:
If data is null why not just pass in null drag target details?

Reply:
This is what I meant by https://github.com/flutter/flutter/pull/133691#pullrequestreview-1668720955

Since the onAcceptWithDetails method is very new, it is easier to change the parameter of the function signature to be nullable. Changing the data in DragTargetDetails to be nullable is a much bigger breaking change since it has been around for a long time.